### PR TITLE
tap_auditor: validate pypi_formula_mappings.json

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Follow up after #16772 to do schema validation and valid formulae checks.

### CODE REVIEW

**Comment by Bo98** [comment]
Note that `json_schemer` is a part of the `pr_upload` gem group so you'll need to add it to `audit` too. See how `rexml` is handled in the Gemfile.

**Comment by cho-m** [comment]
After adding some bad values to JSON, the audit output looks like:
```console
❯ brew audit --tap homebrew/core --only 'json_files,tap_formula_lists'
homebrew/core
  * pypi_formula_mappings.json schema validation failed with following errors:
    * object property at `/aiven-client/foo` is not defined and schema does not allow additional properties
    * value at `/animdl/exclude_packages/1` is not a string
  * pypi_formula_mappings.json references
    formulae or casks that are not found in the homebrew/core tap.
    Invalid formulae or casks: brew
Error: 2 problems in 1 tap detected.
```

I only created parts of JSON Schema needed to run check, but can choose to fill out the other metadata.

Can also consider restricting to `homebrew/core` if there is any chance external taps may be using JSON in other ways.

**Comment by MikeMcQuaid** [comment]
@cho-m love the idea!

**Comment by cho-m** [comment]
Still fine tuning schema. I plan to update `homebrew/core` side to align normalized names similar to how we name resources.

File: Library/Homebrew/data/schemas/pypi_formula_mappings.schema.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        },
+        "additionalProperties": false
+      }
+    ]
+  },
+  "$defs": {
+    "package_name": {
+      "description": "Normalized PyPI package name with optional extras",
+      "type": "string",
+      "pattern": "^[a-z0-9][a-z0-9-]*(\\[[a-z0-9][a-z0-9-]*(,[a-z0-9][a-z0-9-]*)*\\])?$"
```
Comment (cho-m) [inline Library/Homebrew/data/schemas/pypi_formula_mappings.schema.json:47]: This is a simplification of https://packaging.python.org/en/latest/specifications/name-normalization/. It doesn't handle trailing `-` or consecutive `-` as those make the regex messier.

**Pull request closed without merging**